src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -142,12 +143,16 @@ def fuzz(self,
     ])
 
     additional_args.extend(corpus_directories)
-    return self.run_and_wait(
-        additional_args=additional_args,
-        timeout=fuzz_timeout - self.SIGTERM_WAIT_TIME,
-        terminate_before_kill=True,
-        terminate_wait_time=self.SIGTERM_WAIT_TIME,
-        max_stdout_len=MAX_OUTPUT_LEN)
+    try:
+      return self.run_and_wait(
+          additional_args=additional_args,
+          timeout=fuzz_timeout - self.SIGTERM_WAIT_TIME,
+          terminate_before_kill=True,
+          terminate_wait_time=self.SIGTERM_WAIT_TIME,
+          max_stdout_len=MAX_OUTPUT_LEN)
+    finally:
+      # Make sure to unset LD_PRELOAD if we set it to use a plugin.
+      mutator_plugin.unset_mutator_plugin()
```

Comment:
the place where we set this should be closer to where we do this unsetting. As I suggested in a previous comment, can we move the setting (and the unsetting) into the launcher? 

a better solution may not need this unset at all if we e.g. add an argument to the fuzz interface to accept a path to a mutator. then we can just pass the env to the target via the process launching without messing up the parent env. i.e.

```python
def fuzz(self, ...., mutator_path=None):
  if mutator_path:
    env['LD_PRELOAD'] = ...
    ...
    self.run_and_wait(..., env=env, ....)
```

Reply:
I added an extra_env variable to our `new_process` helper functions.